api/v1alpha1/pipeline_types.go

```diff
@@ -53,6 +54,19 @@ const (
 	userPermissionResourceNamespaceLabelAll = "kratix_all_namespaces"
 )
 
+var (
+	kratixSecurityContext = &corev1.SecurityContext{
+		RunAsNonRoot: ptr.To(true),
+		Privileged:   ptr.To(false),
+		Capabilities: &corev1.Capabilities{
+			Drop: []corev1.Capability{"ALL"},
+		},
+		SeccompProfile: &corev1.SeccompProfile{
+			Type: "RuntimeDefault",
+		},
+	}
+)
+
```

Comment:
@catmo-syntasso  Something to think about here:

Do we want the users to be able to set a securityContext for Kratix owned resources (e.g. the containers we add to the pipelines)? In one way, that would make sense as they may have requirements to run containers in their clusters. In another way, that would mean Kratix would be subject to any rules they apply which may or may not be possible with the current code.

@shano and I were discussing and would recommend we stick with a Kratix default (we have picked a sensible one that is default for Kyverno, suggested by Kubernetes docs, etc). This will let us move faster with Kratix without big risk as changing to user provided would be a very small change.